hardcoded.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+// Used to generate data for consts.php INSTALLER_TO_REPO_MINOR_VERSIONS
+// Copy paste in .cow.pat.json from the current release (file is .gitignored)
+
+include 'consts.php';
+
+$fn = '.cow.pat.json';
+if (!file_exists($fn)) {
+    echo "Create $fn\n";
+    die;
+}
+
+$versions = [];
+
+function repoName(string $name)
+{
+    $prefix = '';
+    if (!preg_match('#^(cwp/)#', $name) &&
+        !preg_match('#(/recipe-|/silverstripe-|/comment-notifications)#', $name)
+    ) {
+        $prefix = 'silverstripe-';
+    }
+    return $prefix . explode('/', $name)[1];
+}
+
+function parseNode(string $name, stdClass $node, array &$versions)
+{
+    $repoName = repoName($name);
+    if (!in_array($repoName, LOCKSTEPPED_REPOS) &&
+        !in_array($repoName, NO_INSTALLER_LOCKSTEPPED_REPOS) &&
+        !in_array($repoName, NO_INSTALLER_UNLOCKSTEPPED_REPOS)
+    ) {
+        preg_match('#^([0-9]+\.[0-9]+)#', $node->Version, $m);
+        $versions[$repoName] = preg_replace('#-(alpha|beta|rc)[1-9]+$#', '', $m[1]);
```

Comment:
`#^([0-9]+\.[0-9]+)#` will only match the x.y (e.g. 4.10) part of the version anyway, won't it? So there should be no alpha/beta/etc or even `.0` in `$m[1]`.

Reply:
true, updated